packages/tentales/src/middleware/default-hooks.ts

```diff
@@ -1,24 +1,49 @@
-import { Hook } from "tentales"
+import { Hook, Config } from "tentales"
+import { Log } from "tentales-log"
+import { path } from "ramda"
 import { bodyParserMiddleware } from "./middlewares/body-parser-middleware"
 import { renderMiddleware } from "./middlewares/render-middleware"
 import { fourOhFourMiddleware } from "./middlewares/four-oh-four-middleware"
 import { errorMiddleware } from "./middlewares/error-middleware"
 import { protectServiceRoutesMiddleware } from "./middlewares/protect-service-routes-middelware"
 
-const DEFAULT_HOOKS_PUBLIC: Hook[] = [
-  ["errorMiddleware", [errorMiddleware]],
-  ["bodyParserMiddleware", [bodyParserMiddleware]],
-  ["renderMiddleware", [renderMiddleware]],
-  ["protectServiceRoutesMiddleware", [protectServiceRoutesMiddleware]],
-  ["fourOhFourMiddleware", [fourOhFourMiddleware]],
-]
+const DEFAULT_HOOKS_PUBLIC = (log: Log, config: Config): Hook[] => {
+  const middlewares: Hook[] = [
+    ["errorMiddleware", [errorMiddleware]],
+    ["bodyParserMiddleware", [bodyParserMiddleware]],
+    ["renderMiddleware", [renderMiddleware]],
+  ]
+
+  if (path(["auth", "serverSecret"], config)) {
+    middlewares.push([
+      "protectServiceRoutesMiddleware",
+      [protectServiceRoutesMiddleware],
+    ])
+  } else {
+    log.warn(
+      "Authorization secret not set; no authorization will be carried out.",
+    )
+  }
+
+  middlewares.push(["fourOhFourMiddleware", [fourOhFourMiddleware]])
```

Comment:
Currently we're building a list by first adding some stuff to a list, then conditionally pushing to that list, and then always adding some more stuff. For readability, would it maybe be nice do something along these lines?

```
const middleswares = [
    ["errorMiddleware", [errorMiddleware]],
    ["bodyParserMiddleware", [bodyParserMiddleware]],
    ["renderMiddleware", [renderMiddleware]],
    shouldProtectMiddleware ? ["protectServiceRoutesMiddleware", [protectServiceRoutesMiddleware]] : null,
    ["fourOhFourMiddleware", [fourOhFourMiddleware]]
]
.filter(Boolean)
```

Reply:
Sure!